posts/2022/05/the-evolution-of-the-scipy-developer-cli.md

```diff
@@ -0,0 +1,170 @@
+<!--
+.. title: The evolution of the SciPy developer CLI
+.. slug: the-evolution-of-the-scipy-developer-cli
+.. date: 2022-05-03 8:00:00 UTC+05:30
+.. author: Sayantika Banik
+.. tags: CLI, SciPy, doit, Typer, Rich, Click
+.. category:
+.. link:
+.. description:
+.. type: text
+.. previewimage:
+-->
+
+#### 🤔 What is a command-line interface (CLI)?
+
+Imagine a situation where there is a massive system with various tools and functionality. Every functionality requires a special command or an input from the user, CLI is designed to tackle such situations. Like a catalog it lists all the options present, helping the user to navigate.
```

Comment:
```suggestion
Imagine a situation, where there is a massive system with various tools and functionalities, and where every functionality requires a special command or an input from the user. A CLI is designed to tackle such situations. Like a catalog or menu, it lists all the options available, thus helping the user to navigate a complex system.
```

Reply:
tweaked it a bit for readability mostly. But it's just style, I'm not sure if we have guidelines for comma usage?! So if you don't like it better ignore it. 
I think functionalities should be plural though.